### PR TITLE
`Development`: Add documentation for push notifications with Hermes

### PR DESCRIPTION
**Pull request by sven0311**
### Motivation and Context

This PR adds documentation on how to use the [Artemis iOS client](https://github.com/ls1intum/artemis-ios)

### Description

This PR only adds documentation and does not touch code at all.

### Steps for Testing

1. Take a look at the code documentation.
2. Inspect the Setup Guide -> Hermes section and read the documentation carefully

#### Code Review
- [x] Docu Review 1
- [x] Docu Review 2

### CODE REVIEW

File: docs/dev/setup.rst [hunk 1]
```diff
@@ -600,6 +600,52 @@ instead of the TUM defaults:
 ------------------------------------------------------------------------------------------------------------------------
 
 
+Hermes Service
+--------------
+
+Push notifications for the mobile Android and iOS clients rely on the Hermes_ service.
+To enable push notifications the Hermes service needs to be started separately and special configuration on the Artemis instance is required.
```
Comment (nityanandaz) [inline docs/dev/setup.rst:607]:
```suggestion
To enable push notifications the Hermes service needs to be started separately and the configuration of the Artemis instance must be extended.
```

**Review by nityanandaz — APPROVED** [review]
I understand the documentation :)

File: docs/dev/setup.rst [hunk 2]
```diff
@@ -600,6 +600,52 @@ instead of the TUM defaults:
...
+* ``APNS_CERTIFICATE_PWD``: String - The APNS certificate password
+* ``APNS_PROD_ENVIRONMENT``: Bool - True if it should use the Production APNS Server (Default false)
+
+Furthermore the <APNS_Key>.p12 needs to be mounted into the Docker under the above specified path.
+
+To run the services for Android support the following environment variable is required:
+
+* ``GOOGLE_APPLICATION_CREDENTIALS``: String - Path to the firebase.json
+
+Furthermore the Firebase.json needs to be mounted into the Docker under the above specified path.
```
Comment (terlan98) [inline docs/dev/setup.rst:626]:
```suggestion
Furthermore, the Firebase.json needs to be mounted into the Docker under the above specified path.
```

File: docs/dev/setup.rst [hunk 3]
```diff
@@ -600,6 +600,52 @@ instead of the TUM defaults:
...
+
+To run Hermes you need to clone the `repository <https://github.com/ls1intum/Hermes>`_ and replace the placeholders within the ``docker-compose`` file.
+
+The following environment variables need to be updated for push notifications to Apple devices:
+
+* ``APNS_CERTIFICATE_PATH``: String - Path to the APNs certificate .p12 file as described `here <https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_certificate-based_connection_to_apns>`_
+* ``APNS_CERTIFICATE_PWD``: String - The APNS certificate password
+* ``APNS_PROD_ENVIRONMENT``: Bool - True if it should use the Production APNS Server (Default false)
+
+Furthermore the <APNS_Key>.p12 needs to be mounted into the Docker under the above specified path.
```
Comment (terlan98) [inline docs/dev/setup.rst:620]:
```suggestion
Furthermore, the <APNS_Key>.p12 needs to be mounted into the Docker under the above specified path.
```

File: docs/dev/setup.rst [hunk 4]
```diff
@@ -600,6 +600,52 @@ instead of the TUM defaults:
...
+
+To run the services for Android support the following environment variable is required:
+
+* ``GOOGLE_APPLICATION_CREDENTIALS``: String - Path to the firebase.json
+
+Furthermore the Firebase.json needs to be mounted into the Docker under the above specified path.
+
+To run both APNS and Firebase configure the Environment Variables for both.
```
Comment (terlan98) [inline docs/dev/setup.rst:628]:
```suggestion
To run both APNS and Firebase, configure the environment variables for both.
```

File: docs/dev/setup.rst [hunk 5]
```diff
@@ -600,6 +600,52 @@ instead of the TUM defaults:
...
+Hermes Service
+--------------
+
+Push notifications for the mobile Android and iOS clients rely on the Hermes_ service.
+To enable push notifications the Hermes service needs to be started separately and special configuration on the Artemis instance is required.
+
+Configure and start Hermes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run Hermes you need to clone the `repository <https://github.com/ls1intum/Hermes>`_ and replace the placeholders within the ``docker-compose`` file.
```
Comment (terlan98) [inline docs/dev/setup.rst:612]:
```suggestion
To run Hermes, you need to clone the `repository <https://github.com/ls1intum/Hermes>`_ and replace the placeholders within the ``docker-compose`` file.
```

**Review by terlan98 — COMMENTED** [review]
Some commas are missing, but the content of the documentation looks good 👍

**Review by terlan98 — APPROVED** [review]
Looks good 👍 Thanks for the changes.

**Review by nityanandaz — APPROVED** [review]
Thanks

**Pull request merged**